packages/interactivity/src/store.ts

```diff
@@ -16,8 +16,11 @@ import {
 	resetNamespace,
 } from './hooks';
 
-const isObject = ( item: unknown ): boolean =>
-	!! item && typeof item === 'object' && ! Array.isArray( item );
+const isObject = ( item: unknown ): item is Record< string, unknown > => {
+	if ( typeof item !== 'object' || item === null ) return false;
+	const proto = Object.getPrototypeOf( item );
+	return proto === Object.prototype || proto === null;
```

Comment:
I'm not sure we want to support objects without a prototype. 🤔

The only case I can imagine where a _prototype-less_ object can end up there is by assigning it directly to a property inside `state`, and `deep signal` doesn't proxy that kind of objects. It would be better to return `false` instead.

```suggestion
	return proto === Object.prototype;
```